addons/hr_expense/models/hr_expense.py

```diff
@@ -131,7 +138,10 @@ def _compute_same_currency(self):
     @api.depends('product_id')
     def _compute_product_has_cost(self):
         for expense in self:
-            expense.product_has_cost = bool(expense.product_id and expense.unit_amount)
+            expense.product_has_cost = bool(expense.product_id and expense.product_id.standard_price != 0.0)
```

Comment:
We should use float_compare to avoid rounding issues
```suggestion
            expense.product_has_cost = expense.product_id and float_compare(expense.product_id.standard_price, 0.0, precision_rounding=expense.currency_id.rounding)
```

I didn't test it though

Reply:
Here it's not about currency I think, we just check whether the standard_price is zero or not. We could simply use `precision_digits=2`